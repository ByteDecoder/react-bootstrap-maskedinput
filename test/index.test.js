import React from 'react';
import { mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import ReactBootstrapMaskedInput from '../src';

describe('Component', () => {
  it('renders correctly with basic props', () => {
    const wrapper = mount(
      <ReactBootstrapMaskedInput
        mask='111-111-1111'
      />
    );

    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('passes complex props through to FormControl and MaskedInput', () => {
    const wrapper = mount(
      <ReactBootstrapMaskedInput
        bsClass='my-form-control'
        bsSize='large'
        id='my-id'
        inputRef={ref => ref}
        mask='111-111-1111'
        placeholder='555-555-5555'
        size={10}
      />
    );

    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
