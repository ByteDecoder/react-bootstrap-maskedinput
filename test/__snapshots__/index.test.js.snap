// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component passes complex props through to FormControl and MaskedInput 1`] = `
<MaskedFormControl
  bsClass="my-form-control"
  bsSize="large"
  componentClass="input"
  id="my-id"
  inputRef={[Function]}
  mask="111-111-1111"
  placeholder="555-555-5555"
  size={10}
>
  <FormControl
    bsClass="my-form-control"
    bsSize="large"
    componentClass={[Function]}
    id="my-id"
    inputRef={[Function]}
    mask="111-111-1111"
    placeholder="555-555-5555"
    size={10}
  >
    <MaskedInput
      className="my-form-control input-lg"
      id="my-id"
      mask="111-111-1111"
      placeholder="555-555-5555"
      size={10}
      value=""
    >
      <input
        className="my-form-control input-lg"
        id="my-id"
        mask="111-111-1111"
        maxLength={12}
        onChange={[Function]}
        onKeyDown={[Function]}
        onKeyPress={[Function]}
        onPaste={[Function]}
        placeholder="555-555-5555"
        size={10}
        value=""
      />
    </MaskedInput>
  </FormControl>
</MaskedFormControl>
`;

exports[`Component renders correctly with basic props 1`] = `
<MaskedFormControl
  bsClass="form-control"
  componentClass="input"
  mask="111-111-1111"
>
  <FormControl
    bsClass="form-control"
    componentClass={[Function]}
    mask="111-111-1111"
  >
    <MaskedInput
      className="form-control"
      mask="111-111-1111"
      value=""
    >
      <input
        className="form-control"
        mask="111-111-1111"
        maxLength={12}
        onChange={[Function]}
        onKeyDown={[Function]}
        onKeyPress={[Function]}
        onPaste={[Function]}
        placeholder="___-___-____"
        size={12}
        value=""
      />
    </MaskedInput>
  </FormControl>
</MaskedFormControl>
`;
